src/main/java/com/google/cloud/discotoproto3converter/proto3/DocumentToProtoConverter.java

```diff
@@ -127,16 +128,26 @@ private void cleanupEnumNamingConflicts() {
       }
       if (enumFieldsCount > enumFieldsNames.size()) {
         message.getEnums().clear();
-        ListIterator<Field> fields = message.getFields().listIterator();
-        while (fields.hasNext()) {
-          Field f = fields.next();
+        SortedSet<Field> newFields = new TreeSet<>();
```

Comment:
nit: why call this `newFields` and not simply `fields` again? (I get it's probably less confusing doing this PR, but for submitted code it may be clearer to not have the "new"---or convey what it means.

Reply:
Remind me: this function converts all enums to strings in case of a name conflict, but that doesn't happen for compute?

Reply:
We talked offline and you said this happens for some fields that are not enums in the backend, but are exposed as enums in the Discovery file, and then may have name collisions causing us to revert them back to strings. Could you provide more information (here or internally) on which Discovery-enums these are and how you arrive at the conclusion they are unlikely to change in the future? We want to have clear outlook for when breaking changes may happen. Thanks!

Reply:
For compute this kind of logic gets applied for 5 messages in total
```
Condition
InstanceGroupManagerUpdatePolicy
InstanceGroupManagersApplyUpdatesRequest
RegionInstanceGroupManagersApplyUpdatesRequest
UpdateInstanceRequest
```

This logic is applied if there is a naming conflict in the enums.

The reason why it is applied - in protobuf enums can't have naming conflicts, and if those were actual enums on server side, the could not be defined the way they are defined, that is why they must be strings. 

I can confirm that this list of 5 affected messages has been like this for more than a year at least (so it is not something which changes frequently).

Reply:
OK, but this could cause a problem when it does get updated. We should think about this (separately).